### PR TITLE
feat(azure): Log if we cannot find a reviewer

### PR DESCRIPTION
**Pull request by jeley1000**
## Changes

Log a warning for any configured reviewers that cannot be found in Azure DevOps.

## Context

Closes #26735 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
needs a test

**Comment by jeley1000** [comment]
> needs a test

Added some tests and fixed a bug! @viceice

File: lib/modules/platform/azure/index.ts [hunk 1]
```diff
@@ -855,6 +855,7 @@ async function getUserIds(users: string[]): Promise<User[]> {
   const repos = await azureApiGit.getRepositories();
   const repo = repos.filter((c) => c.id === config.repoId)[0];
   const requiredReviewerPrefix = 'required:';
+  const validReviewers: string[] = [];
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:858]:
```suggestion
  const validReviewers = new Set<string>();
```
use a set

File: lib/modules/platform/azure/index.ts [hunk 2]
```diff
@@ -890,6 +891,8 @@ async function getUserIds(users: string[]): Promise<User[]> {
               name: reviewer,
               isRequired,
             });
+
+            validReviewers.push(reviewer);
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:895]:
```suggestion
            validReviewers.add(reviewer);
```

File: lib/modules/platform/azure/index.ts [hunk 3]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
         if (ids.filter((c) => c.id === t.id).length === 0) {
           // TODO #22198
           ids.push({ id: t.id!, name: reviewer, isRequired });
+
+          validReviewers.push(reviewer);
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:915]:
```suggestion
          validReviewers.add(reviewer);
```

File: lib/modules/platform/azure/index.ts [hunk 4]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
...
         }
       }
     });
   });
 
+  users.forEach((u) => {
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:921]:
```suggestion
  for (const u of users) {
```
use for-of loop

File: lib/modules/platform/azure/index.ts [hunk 5]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
...
+    let reviewer = u.replace(requiredReviewerPrefix, '');
+    if (validReviewers.indexOf(reviewer) == -1) {
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:923]:
```suggestion
    if (validReviewers.has(reviewer)) {
```

File: lib/modules/platform/azure/index.ts [hunk 6]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
...
+      logger.warn(`${reviewer} is neither an Azure DevOps Team nor a user associated with a Team`);
+    }
+  });
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:926]:
```suggestion
  };
```

File: lib/modules/platform/azure/index.ts [hunk 7]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
...
           ids.push({ id: t.id!, name: reviewer, isRequired });
+
+          validReviewers.push(reviewer);
         }
       }
     });
   });
 
+  users.forEach((u) => {
+    let reviewer = u.replace(requiredReviewerPrefix, '');
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:922]:
```suggestion
    const reviewer = u.replace(requiredReviewerPrefix, '');
```

File: lib/modules/platform/azure/index.ts [hunk 8]
```diff
@@ -908,11 +911,20 @@ async function getUserIds(users: string[]): Promise<User[]> {
...
+    if (validReviewers.indexOf(reviewer) == -1) {
+      logger.warn(`${reviewer} is neither an Azure DevOps Team nor a user associated with a Team`);
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:924]:
```suggestion
      logger.once.info(`${reviewer} is neither an Azure DevOps Team nor a user associated with a Team`);
```
I think warn is too hard here. 

@rarkins WDYT?

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by viceice** [comment]
you need to fix your commit author and editor to fulfill the cla check

File: lib/modules/platform/azure/index.spec.ts [hunk 9]
```diff
@@ -1530,6 +1530,33 @@ describe('modules/platform/azure/index', () => {
       );
       await azure.addReviewers(123, ['test@bonjour.fr', 'jyc', 'required:def']);
       expect(azureApi.gitApi).toHaveBeenCalledTimes(3);
+      expect(logger.warn).toHaveBeenCalledTimes(1);
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:1533]: did you forget to update tests?

Comment (jeley1000) [reply lib/modules/platform/azure/index.spec.ts]: Fixed to check for the logger.once.info instead. Thank you

**Review by viceice — APPROVED** [review]

**Comment by ozziepeeps** [comment]
How do we get this merged @viceice?

**Pull request merged**